k8s/src/test/scala/io/buoyant/k8s/CustomResourceTest.scala

```diff
@@ -14,8 +14,8 @@ import org.scalatest.{FunSuite, OptionValues}
  * This test both exercises the versioning functionality in `io.buoyant.k8s`, and provides
  * an example implementation. (See `namerd` for the current usage in the Buoyant suite of projects).
  */
-class ThirdPartyResourceTest extends FunSuite with Awaits with OptionValues {
-  import ThirdPartyResourceTest._
+class CustomResourceTest extends FunSuite with Awaits with OptionValues {
+  import CustomResourceTest._
```

Comment:
I don't think it's correct to replace the existing `ThirdPartyResourceTest` with a new `CustomResourceTest`, since we will need to continue supporting third-party resources as well in order to remain backwards-compatible. I think that rather than replacing the test, we ought to have separate tests for TPR and CRD.